solr/test-framework/src/java/org/apache/solr/BaseDistributedSearchTestCase.java

```diff
@@ -606,6 +608,10 @@ protected QueryResponse query(boolean setDistribParams, SolrParams p) throws Exc
     final QueryResponse controlRsp = controlClient.query(params);
     validateControlData(controlRsp);
 
+    if (shardCount == 0) {//mostly for temp debugging
+      return controlRsp;
+    }
+
```

Comment:
this supposed to land?

Reply:
yes and the next diff too.  This allows you to set numShards=0 temporarily for debugging and and thus only have the "control client".  Granted it defeats the purpose of using BaseDistributedSearchTestCase but it's only meant to temporarily remove distracting aspects while some bug is being worked out that is not related to distributed search.